File: src/main/cpp/color.cpp
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "color.h"

namespace pulse {

const std::array<int, Color::VALUES_SIZE> Color::values = {
  WHITE, BLACK
};

bool Color::isValid(int color) {
  switch (color) {
    case WHITE:
    case BLACK:
      return true;
    default:
      return false;
  }
}

int Color::opposite(int color) {
  switch (color) {
    case WHITE:
      return BLACK;
    case BLACK:
      return WHITE;
    default:
      throw std::exception();
  }
}

}

File: pulse-java/src/main/java/com/fluxchess/pulse/java/model/Color.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.java.model;

public final class Color {

	public static final int WHITE = 0;
	public static final int BLACK = 1;

	public static final int NOCOLOR = 2;

	public static final int[] values = {
		WHITE, BLACK
	};

	private Color() {
	}

	public static int opposite(int color) {
		switch (color) {
			case WHITE:
				return BLACK;
			case BLACK:
				return WHITE;
			default:
				throw new IllegalArgumentException();
		}
	}
}
